### DISCUSSION TITLE
Custom component with extended type does not update content value

### DISCUSSION THREAD

**Discussion by ptt-joaoferreira** [GrapesJS/grapesjs#5823] [Q&A]
Hello,

I'm trying to create a new component by extending the "core" type **text** so I am abre to freely edit text on canvas but using my custom traits.
When I update the textbox in canvas, the HTML get's updates aswell.
When I use in traits `...getSelected().set('content', 'whatever); `the value in canvas disappears and html doesn't get updated.
This behaviour just happens when I am extending in the new type. If I remove that extend, the canvas works as expected but, I am unable to edit the text.

Any ideas on this?

Thanks

**Comment by artf** [comment]
don't use `content`, update directly the inner components

**Reply by ptt-joaoferreira** [reply]
But in this case, I don't have any components inside the compontent itself. Unless that, when you extend, you pass an inner component to the current one, wich find does not catch.

`domComponents.addType("txt", {
        extend: "text",
        model: {
            defaults: {
              (...)`

whis is how I am extending.

**Reply by ptt-joaoferreira** [reply]
@artf sorry to tag you but I really could use some help

**Reply by artf** [reply]
What I'm suggesting is doing this `component.components('whatever')` instead of `component.set('content', 'whatever')`
